tiledb/sm/filesystem/uri.cc

```diff
@@ -211,29 +211,59 @@ bool URI::is_tiledb() const {
 }
 
 Status URI::get_rest_components(
-    std::string* array_namespace, std::string* array_uri) const {
+    std::string* array_namespace, std::string* array_uri, bool legacy) const {
   const std::string prefix = "tiledb://";
+  const std::string ns_component =
+      legacy ? "tiledb://<namespace>" : "tiledb://<workspace>/<teamspace>";
   const auto error_st = Status_RestError(
-      "Invalid array URI for REST service; expected format is "
-      "'tiledb://<namespace>/<array-name>' or "
-      "'tiledb://<namespace>/<array-uri>'.");
+      "Invalid array URI for REST service; expected format is " + ns_component +
+      "/<array-name>' or " + ns_component + "/<array-uri>'.");
 
   if (!is_tiledb() || uri_.empty() || uri_.find(prefix) == std::string::npos ||
-      uri_.size() <= prefix.size())
+      uri_.size() <= prefix.size()) {
     return LOG_STATUS(error_st);
+  }
 
-  // Find '/' between namespace and array uri.
-  auto slash = uri_.find('/', prefix.size());
-  if (slash == std::string::npos)
-    return LOG_STATUS(error_st);
+  if (legacy) {
+    // Extract '<namespace>' if we are using a 3.0 REST server.
```

Comment:
Let's remove all references to 3.0 as they'll soon become stale (when the new version comes). Let's replace with "TileDB Server"

Reply:
Done